rust/lancedb/src/table.rs

```diff
@@ -978,9 +978,17 @@ impl TableInternal for NativeTable {
             Select::All => { /* Do nothing */ }
         }
 
-        query.filter.as_ref().map(|f| scanner.filter(f));
-        query.refine_factor.map(|rf| scanner.refine(rf));
-        query.metric_type.map(|mt| scanner.distance_metric(mt));
+        if let Some(filter) = &query.filter {
+            scanner.filter(filter)?;
+        }
+
+        if let Some(refine_factor) = query.refine_factor {
+            scanner.refine(refine_factor);
+        }
+
+        if let Some(metric_type) = query.metric_type {
+            scanner.distance_metric(metric_type);
+        }
```

Comment:
I found that by changing to `if let Some(...) = ...`, then clippy will warn us about unused results. So to be defensive, I've changed these other parameters as well.